### ISSUE TITLE
c.unwrap(classOf[OracleConnection]) fails when logSql=true (ClassCastException)

### ISSUE THREAD

**Issue by arlowhite** [playframework/playframework#6471]
### Environment

- Play 2.5.4
- scalaVersion "2.11.8"
- Anorm 2.5.2
- JDK Oracle 1.8.0_102
- Oracle JDBC 
 - Implementation-Version: 11.2.0.3.0
 - Repository-Id: JAVAVM_11.2.0.4.0_LINUX.X64_130711

### Bug

This will work when **logSql=false**, but fails when **logSql=true**
```
db.withConnection { implicit c  =>
  c.unwrap(classOf[OracleConnection])
}
```

When **logSql=true**, I get this exception:

> com.sun.proxy.$Proxy17 cannot be cast to oracle.jdbc.OracleConnection 
> java.lang.ClassCastException: com.sun.proxy.$Proxy17 cannot be cast to oracle.jdbc.OracleConnection

### Notes

This happens with both Hikari and Bone connection pool.
I think **ConnectionPool.scala** needs to be fixed, but I'm not sure how:
https://github.com/playframework/playframework/blob/d6c2673d91d85fd37de424951ee5ad9f4f4cce98/framework/src/play-jdbc/src/main/scala/play/api/db/ConnectionPool.scala#L86

db.default.driver=oracle.jdbc.driver.OracleDriver
db.default.url="jdbc:oracle:thin:@mydbhost.com:1521:MYDBSID"

**Comment by gmethvin** [comment]
We use [jdbcdslog](https://github.com/jdbcdslog/jdbcdslog) to implement the logging, and that uses a dynamic proxy to do the logging rather than the `Wrapper` interface implemented by `Connection` that allows you to retrieve the wrapped instance via `unwrap`. So it seems like it's a limitation of that library that should ideally be fixed there.

**Comment by arlowhite** [comment]
Ok, do you want me to create a GitHub issue on their project?

Maybe Play could throw an Exception when you call unwrap with logSql=true

> "Connection cannot be unwrapped while using logSql "

**Comment by gmethvin** [comment]
> Ok, do you want me to create a GitHub issue on their project?

Yeah. I'm not sure if they consider it a bug, but it's at least worth having the discussion.

> Maybe Play could throw an Exception when you call unwrap with logSql=true

That's not so simple since we're basically delegating everything to the jdbcdslog wrapper right now.

**Comment by schmitch** [comment]
actually the version of jdbcdslog we are using is not maintained.

**Comment by gmethvin** [comment]
@schmitch There's nothing to indicate it's not being maintained right now, though: https://github.com/jdbcdslog/jdbcdslog

Last commit was Feb 26 and we're using the latest version recommended there.

I suppose if we have any doubts we can just fork it or use a different library.

**Comment by schmitch** [comment]
we are using: 
https://github.com/playframework/playframework/blob/master/framework/project/Dependencies.scala#L46
which was released on December 2011

and from the repo:

> jdbcdslog-exp is an experimental fork of jdbcdslog

and if you look at the commit graph you will see that there wasn't much than, expect README.md changes. And some minor changes.

**Comment by gmethvin** [comment]
We should consider using an alternate library for logging that is more well-maintained, or write our own.
